ref: Remove explore endpoint and refactor

samuveth opened the pull request:
### Issues

#3863 
#1437 


### Changes

1. Replace `explore` endpoint and sorting logic in `useSpaces` with a `GraphQL` query
2. Implement two separate queries:
   - One for the home page and ranking page
   - One for general use
3. Introduce a `Space` type for use with Spaces from the new query
4. Refactor multiple components to accommodate the different data structure from the new endpoint
5. Update `loadExtendedSpaces` -> `loadExtendedSpace` query:
   - Now loads a single space
   - Continues to cache the data


### Things to test

1. Test `/` page spaces searching and filtering functionality
2. Test `/ranking` page functionality 
3. Test flagged spaces and proposals
4. Test if hidden spaces are visible on homepage
6. Test timeline `all` should only show proposals from verified spaces 


### To-Do


- [x] Fix ranking
- [x] Fix aggregate count
- [x] Fix category count
- [x] Fix string search
- [x] Fix infinite scroll
- [x] Fix active proposal count sidebar
- [x] Change to use production hub endpoint
- [x] Test flagged spaces and proposals once using production hub
- [x] Run E2E tests once using production hub
- [x] Test search for hidden space by exact space id




### Self-review checklist
- [x] I have performed a full self-review of my changes
- [x] I have tested my changes on a preview deployment
- [x] I have tested my changes on a custom domain
- [x] I have run end-to-end tests `yarn cypress:test:e2e`, and they have passed

### Additional notes or considerations

Todmy commented:
I have noticed that while going back from any space page, a user will be retrieved to the home page with the initial state where we have only 12 items loaded. On one side It's not convenient if I scrolled a couple of times to get to some spaces and on going back I need to scroll again. But from another point of view, if we decide to save the scrolling state: if I have scrolled till the end of the home page, on going back we need to load all available spaces.
Have you thought about this behavior?

j0hnfl0w commented:
> I have noticed that while going back from any space page, a user will be retrieved to the home page with the initial state where we have only 12 items loaded. On one side It's not convenient if I scrolled a couple of times to get to some spaces and on going back I need to scroll again. But from another point of view, if we decide to save the scrolling state: if I have scrolled till the end of the home page, on going back we need to load all available spaces. Have you thought about this behavior?

We can do it with <keep-alive /> but i think we can do it later.
Its like a tweeter feed, when after closing an app you return to the same tweet (sometimes it is not work properly)
wdyt @samuveth ?

samuveth commented on src/helpers/apollo.ts:
```diff
@@ -8,7 +8,7 @@ import {
 const httpLink = createHttpLink({
   // You should use an absolute URL here
-  uri: `${import.meta.env.VITE_HUB_URL}/graphql`
+  uri: `https://sea-turtle-app-hdjod.ondigitalocean.app/graphql`
```
Yep, will change it once changes on the HUB were merged.

samuveth commented on src/composables/useSpaces.ts:
```diff
@@ -1,106 +1,112 @@
     );
+  }
+
+  async function loadSpacesHome(variables: any = {}) {
```
Not convinced, I think having them separated makes the code easier to read.

samuveth commented:
Yes, we can do that by moving the state outside of the composable. Using keep alive will not work because going to a space involved changing the route.

samuveth commented:
Changes since last review:
1. Remove use of lists verified.json, flagged proposal urls & ids, and use `verified` & `flagged` from query instead
2. Use `KeepAlive` instead of global state (which was causing issues). @j0hnfl0w I didn't know it was possible to use `KeepAlive` with `RouterView` like [this](https://github.com/snapshot-labs/snapshot/blob/8cd9bfb561b434e2ba41703d87dc03e8de069dd7/src/App.vue#L44).
3. Refactor `BaseSearch`, remove handling of route query to keep the component simple

Todmy reviewed (approved):
tACK

ChaituVR commented on src/helpers/apollo.ts:
```diff
@@ -8,7 +8,7 @@ import {
 const httpLink = createHttpLink({
   // You should use an absolute URL here
-  uri: `${import.meta.env.VITE_HUB_URL}/graphql`
+  uri: `https://sea-turtle-app-hdjod.ondigitalocean.app/graphql`
```
```suggestion
  uri: `${import.meta.env.VITE_HUB_URL}/graphql`
```
